src/main/kotlin/nl/avisi/structurizr/site/generatr/site/views/Markdown.kt

```diff
@@ -66,3 +72,18 @@ private class CustomLinkResolver(private val pageViewModel: PageViewModel) : Lin
         override fun affectsGlobalScope() = false
     }
 }
+
+private fun Element.transformEmbeddedDiagramElements(pageViewModel: PageViewModel) = this.allElements
+    .toList()
+    .filter { it.tag().name == "img" && it.attr("src").startsWith("embed:") }
+    .forEach {
+        val diagramId = it.attr("src").substring(6)
+        val svg = File(svgPathname(pageViewModel.generatorContext, diagramId)).readText()
```

Comment:
We could also move the File loading to the model site, but therefore we would need to parse the markdown twice. Another way, may be the cleanest one, would be to just pass a loading function into the view (similar to the `hrefFactory` used in `TableViewModels`).

Reply:
Loading of generated SVG's is now further in the background.